src/transformers/pipelines.py

```diff
@@ -1175,6 +1176,92 @@ def span_to_answer(self, text: str, start: int, end: int):
         return {"answer": " ".join(words), "start": max(0, char_start_idx), "end": min(len(text), char_end_idx)}
 
 
+class SummarizationPipeline(Pipeline):
+    """
+    Summarize news articles and other documents
+
+    Usage::
+
+        summarizer = pipeline("summarization")
+        summarizer("Sam Shleifer writes the best docstring examples in the whole world.")
+
+    Supported Models:
+        The models that this pipeline can use are models that have been fine-tuned on a summarization task, which is
+        currently only ``BartForMaskedLM.from_pretrained('bart-large-cnn')``
+
+    Arguments:
+        model (:obj:`str` or :obj:`~transformers.PreTrainedModel` or :obj:`~transformers.TFPreTrainedModel`, `optional`, defaults to :obj:`None`):
+            The model that will be used by the pipeline to make predictions. This can be :obj:`None`, a string
+            checkpoint identifier or an actual pre-trained model inheriting from
+            :class:`~transformers.PreTrainedModel` for PyTorch and :class:`~transformers.TFPreTrainedModel` for
+            TensorFlow.
+
+            If :obj:`None`, the default of the pipeline will be loaded.
+        tokenizer (:obj:`str` or :obj:`~transformers.PreTrainedTokenizer`, `optional`, defaults to :obj:`None`):
+            The tokenizer that will be used by the pipeline to encode data for the model. This can be :obj:`None`,
+            a string checkpoint identifier or an actual pre-trained tokenizer inheriting from
+            :class:`~transformers.PreTrainedTokenizer`.
+
+            If :obj:`None`, the default of the pipeline will be loaded.
+        modelcard (:obj:`str` or :class:`~transformers.ModelCard`, `optional`, defaults to :obj:`None`):
+            Model card attributed to the model for this pipeline.
+        framework (:obj:`str`, `optional`, defaults to :obj:`None`):
+            The framework to use, either "pt" for PyTorch or "tf" for TensorFlow. The specified framework must be
+            installed.
+
+            If no framework is specified, will default to the one currently installed. If no framework is specified
+            and both frameworks are installed, will default to PyTorch.
+        args_parser (:class:`~transformers.pipelines.ArgumentHandler`, `optional`, defaults to :obj:`None`):
+            Reference to the object in charge of parsing supplied pipeline parameters.
+        device (:obj:`int`, `optional`, defaults to :obj:`-1`):
+            Device ordinal for CPU/GPU supports. Setting this to -1 will leverage CPU, >=0 will run the model
+            on the associated CUDA device id.
+    """
+
+    task = "summarization"
+
+    def __call__(
+        self,
+        *texts,
+        return_tensors=False,
+        return_text=True,
+        num_beams=4,
+        length_penalty=2.0,
+        max_length=140,
+        min_len=20,
+        no_repeat_ngram_size=3
```

Comment:
Do we really want to expose all those options? or a minimal subset? (the other pipelines do not expose any option)

Reply:
That's a good point.
I feel like selecting just a subset of the generate method arguments is a bit arbitrary.

What about something like what we usually do:
- good defaults for people who just want something to work out-of-the-box (like the one here), and
- full customizability for the others, e.g. by having `generate_kwargs` that is directly transmitted to the `model.generate()` method and linking to the `generate()` method doc in the the docstring for the list of the arguments (that will likely evolve in the future as better decoding mechanisms are designed).

Reply:
Love the idea of passing through `generate_kwargs`.
In terms of exposing options, I was preparing to support different kwargs for `bart-large-xsum`, bc the author wrote "Use beam=6, lenpen=1.0, max_len_b=60, min_len=10 for Xsum Generation" in  https://github.com/pytorch/fairseq/blob/master/examples/bart/README.summarization.md#L121

Reply:
That's also a good point. We don't want to expose two different pipelines for two trained models.

Here is what we should do if it's not already the case: all the parameters of the `generate()` method are actually stored in the configuration of the models (or more precisely, they wil be once #3140 is merged) so you should update the configuration of the two pretrained summarization weights to have the best decoding parameters as selected by the authors.

cc @patrickvonplaten since it's related to his work